### PR TITLE
Adds a GeoIP subsystem

### PR DESCRIPTION
**Pull request by ghost**
:cl:
admin: Admins can now see the country a user is in!
/:cl:

Requires a SQL update:
```
DROP TABLE IF EXISTS `ipstack`;
CREATE TABLE `ipstack` (
	`ip` int(10) unsigned NOT NULL,
	`country` varchar(32) NOT NULL,
	PRIMARY KEY (`ip`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```

### CODE REVIEW

File: hippiestation/code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -1,6 +1,12 @@
 /client/New()
 	. = ..()
 	mentor_datum_set()
+	if(CONFIG_GET(string/ipstack_api_key))
+		country = SSipstack.check_ip(address)
+		if(country == "???")
+			message_admins("<span class='adminnotice'>GeoIP for [key_name_admin(src)] was invalid!</span>")
+		else if(country == "Brazil")
+			message_admins("<span class='adminnotice'>[key_name_admin(src)] is a Brazillian!</span>")
```
Comment (Manly-Dwarf) [inline hippiestation/code/modules/client/client_procs.dm]: nice spelling error steamp0rt you are ugly

**Comment by ghost** [comment]
i forgot this also requires a config update

`IPSTACK_API_KEY` needs to be set.

**Comment by Maroonenahmoron** [comment]
This is a brilliant idea, an attack plan for rogue russians and brazillians trying to storm us

**Review by yoyobatty — APPROVED** [review]
Code looks fine, I'll get this merged, update config and get Gregg or pyko to update the db.

**Review by yoyobatty — APPROVED** [review]

**Pull request merged**